lib/msf/jmx/mbean/server_connection.rb

```diff
@@ -0,0 +1,153 @@
+          stream.contents << Rex::Java::Serialization::Model::NullReference.new
+          stream.contents << Rex::Java::Serialization::Model::NullReference.new
+
+          stream
+        end
+
+        # Builds a Rex::Java::Serialization::Model::Stream to simulate a call to the
+        # Java getObjectInstance method.
+        #
+        # @param opts [Hash{Symbol => String}]
+        # @option opts [String] :obj_id the jmx endpoint ObjId
+        # @option opts [String] :name the name of the MBean
+        # @return [Rex::Java::Serialization::Model::Stream]
+        def get_object_instance_stream(opts = {})
+          obj_id = opts[:obj_id] || "\x00" * 22
+          name = opts[:name] || ''
+
+          builder = Rex::Java::Serialization::Builder.new
+
+          block_data = Rex::Java::Serialization::Model::BlockData.new(nil, "#{obj_id}\xff\xff\xff\xff\x60\x73\xb3\x36\x1f\x37\xbd\xc2")
+
+          new_object = builder.new_object(
+            name: 'javax.management.ObjectName',
+            serial: 0xf03a71beb6d15cf,
+            flags: 3
+          )
+
+          stream = Rex::Java::Serialization::Model::Stream.new
+          stream.contents << block_data
+          stream.contents << new_object
+          stream.contents << Rex::Java::Serialization::Model::Utf.new(nil, name)
+          stream.contents << Rex::Java::Serialization::Model::EndBlockData.new
+          stream.contents << Rex::Java::Serialization::Model::NullReference.new
+
+          stream
+        end
+
+        # Builds a Rex::Java::Serialization::Model::Stream to simulate a call
+        # to the Java invoke method.
+        #
+        # @param opts [Hash{Symbol => String}]
+        # @option opts [String] :obj_id the jmx endpoint ObjId
+        # @option opts [String] :object the object whose method we want to call
+        # @option opts [String] :method the method name to invoke
+        # @option opts [String] :args the arguments of the method to invoke
+        # @return [Rex::Java::Serialization::Model::Stream]
+        def invoke_stream(opts = {})
+          obj_id = opts[:obj_id] || "\x00" * 22
+          object_name = opts[:object] || ''
+          method_name = opts[:method] || ''
+          arguments = opts[:args] || {}
+          builder = Rex::Java::Serialization::Builder.new
+
+          block_data = Rex::Java::Serialization::Model::BlockData.new(nil, "#{obj_id}\xff\xff\xff\xff\x13\xe7\xd6\x94\x17\xe5\xda\x20")
+
+          new_object = builder.new_object(
+            name: 'javax.management.ObjectName',
+            serial: 0xf03a71beb6d15cf,
+            flags: 3
+          )
+
+          data_binary = builder.new_array(
+            name: '[B',
+            serial: 0xacf317f8060854e0,
+            values_type: 'byte',
+            values: invoke_arguments_stream(arguments).encode.unpack('C*')
+          )
+
+          marshall_object = builder.new_object(
+            name: 'java.rmi.MarshalledObject',
+            serial: 0x7cbd1e97ed63fc3e,
```

Comment:
What is the significance of these serial numbers?   How are you calculating them?